core/controllers/reader.py

```diff
@@ -218,18 +148,13 @@ def classify(exp_id, state, answer, params):
     interaction_instance = interaction_registry.Registry.get_interaction_by_id(
         state.interaction.id)
     normalized_answer = interaction_instance.normalize_answer(answer)
+    response = None
 
-    fs = fs_domain.AbstractFileSystem(fs_domain.ExplorationFileSystem(exp_id))
-    input_type = interaction_instance.answer_type
-
-    response = classify_hard_rule(
-        state, params, input_type, normalized_answer, fs)
-    if response is None:
-        response = classify_soft_rule(
-            state, params, input_type, normalized_answer, fs)
     if (interaction_instance.is_string_classifier_trainable and
-            feconf.ENABLE_STRING_CLASSIFIER and response is None):
+            feconf.ENABLE_STRING_CLASSIFIER):
         response = classify_string_classifier_rule(state, normalized_answer)
+    elif feconf.ENABLE_STRING_CLASSIFIER:
```

Comment:
hmm yeah good point. I think we can actually get rid of ENABLE_STRING_CLASSIFIER in the backend, since the frontend checks.

or should we still do a backend check?